evalml/automl/automl_search.py

```diff
@@ -185,13 +188,25 @@ def __init__(self,
         else:
             raise TypeError("max_time must be a float, int, or string. Received a {}.".format(type(max_time)))
 
+        if _max_batches:
+            if not max_batches:
+                max_batches = _max_batches
+            logger.warning("`_max_batches` will be deprecated in the next release. Use `max_batches` instead.")
+
+        if max_batches is not None and max_batches <= 0:
+            raise ValueError(f"Parameter max batches must be None or non-negative. Received {max_batches}.")
+        self.max_batches = max_batches
+        # This is the default value for IterativeAlgorithm - setting this explicitly makes sure that
+        # the behavior of max_batches does not break if IterativeAlgorithm is changed.
+        self._pipelines_per_batch = 5
```

Comment:
Could we add `_pipelines_per_batch` as an automl search parameter as well? I do think we should keep that one private while we're getting parallel support in place, and then once we're confident we wanna keep it we can consider making it public. But it'll be nice to have external control over this for perf testing purposes. Ok to do this separately, or file a separate issue.

Reply:
Sure. I'll file this as a separate issue.

Reply:
Thanks!